ADQL.tex

```diff
@@ -126,11 +122,13 @@ \subsection{Extended functionality}
 % Explicit permisson for services that provide additional functionality.
 This document defines the minimum set of functions, operators and datatypes
 that a service MUST implement in order to register as a service that
-implements this version of the ADQL specification.
+implements this version of the ADQL specification. It also defines optional
+features that a service MUST implement as such if it requires some or all of
+them.
```

Comment:
It also defines optional features that a service ~~MUST~~ **MAY**  implement ~~as such~~ if it requires ~~some or all of~~ them.

Reply:
They aren't optional if it MUST implement them.

Reply:
I think the intention is that if they are implemented at all, they have to be implemented as described in this specification ("as such").

Reply:
Actually my "MUST" was on the "as such". I meant that if a feature (e.g. put a string in lower case) is desired, it MUST be implemented as specified in the ADQL document (e.g. the function LOWER). But maybe it should rather be a "SHOULD" or something like "it is strongly encouraged to implement the corresponding optional ADQL feature, if existing".

That may be too difficult to write (at least, it is for me, with my poor english :-S ). And this is probably asking too much as well :/

Reply:
> I think the intention is that if they are implemented at all, they have to be implemented as described in this specification ("as such").

Yep. That's what I meant :D

Reply:
Ok, sorry, that wasn't clear to me. Suggest a long form to be explicit?
"If a service does provide one of the optional features, it MUST be implemented according to this specification."